### PR TITLE
fix: Disable saving threshold check if no threshold selected

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes https://github.com/influxdata/influxdb/issues/15334

Describe your proposed changes here.
Disable saving threshold check if no threshold is selected

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/alerting/components/CheckEOHeader.tsx [hunk 1]
```diff
@@ -98,6 +98,10 @@ const CheckEOHeader: FC<Props> = ({
   }
 
   const {singleField, singleAggregateFunc} = isDraftQueryAlertable(draftQueries)
+  const oneOrMoreThresholds =
+    check.type === 'threshold'
+      ? check.thresholds && !!check.thresholds.length
```
Comment (hoorayimhelping) [inline ui/src/alerting/components/CheckEOHeader.tsx]: do we prefer `!!` over `Boolean()`?

File: ui/src/alerting/components/CheckEOHeader.tsx [hunk 2]
```diff
@@ -98,6 +98,10 @@ const CheckEOHeader: FC<Props> = ({
...
+      : null
```
Comment (hoorayimhelping) [inline ui/src/alerting/components/CheckEOHeader.tsx]: i think this should be `false`, not `null`. `CheckEOSaveButton` is expecting `oneOrMoreThresholds` to be a boolean and making this explicitly `false` will let us not rely on JS's sometimes confusing type casting. what do you think?

File: ui/src/alerting/components/CheckEOHeader.tsx (hunk 1 above)
Comment (hoorayimhelping) [inline ui/src/alerting/components/CheckEOHeader.tsx]: do we prefer `!!` over `Boolean()`?

Comment (ebb-tide) [reply ui/src/alerting/components/CheckEOHeader.tsx]: no strong preference but `!!` feels more readable?

**Pull request merged**